mtp_api/apps/security/views_admin.py

```diff
@@ -0,0 +1,98 @@
+import collections
+import itertools
+import logging
+
+from django.db import models
+from django.db.models.functions import TruncMonth
+from django.utils.translation import gettext_lazy as _
+
+from core.forms import BasePeriodAdminReportForm
+from core.views import BaseAdminReportView
+from credit.models import Credit
+from security.models import Check
+
+logger = logging.getLogger('mtp')
+
+
+class CheckReportForm(BasePeriodAdminReportForm):
+    pass
+
+
+class CheckReportAdminView(BaseAdminReportView):
+    title = _('Check report (for FIU)')
+    template_name = 'admin/security/check/report.html'
+    form_class = CheckReportForm
+
+    def get_context_data(self, **kwargs):
+        context_data = super().get_context_data(**kwargs)
+        form = context_data['form']
+
+        current_period = form.current_period
+        format_date = form.period_formatter
+
+        rows = form.group_months_into_periods(self.get_monthly_rows())
+        rows = list(filter(lambda r: r['date'] < current_period, rows))
+        for row in rows:
+            row['date_label'] = format_date(row['date'])
+        context_data['rows'] = rows
+
+        context_data['opts'] = Check._meta
+        context_data['form'] = form
+        return context_data
+
+    def get_monthly_rows(self):
+        """
+        Gets all credit count and amounts grouped by month and categorised by security check status.
+        """
+        # subquery used to determine if a credit's security check triggered any rules
+        rules_triggered = Check.objects.filter(credit=models.OuterRef('pk'), rules__len__gte=1).values('pk')
+
+        queryset = (
+            # given all credits
+            Credit.objects_all
+            # take only credits since FIU checks went live
+            # take only credits with a check (i.e. debit card payment was completed)
+            .filter(created__date__gte='2020-01-02', security_check__status__isnull=False)
+            # remove ordering
+            .order_by()
```

Comment:
you do need it because the Credit model has a default ordering. having a default ordering makes aggregates (the `.values` / `.annotate` combo) not work correctly

Reply:
https://docs.djangoproject.com/en/2.2/topics/db/aggregation/#interaction-with-default-ordering-or-order-by

Reply:
That's sneaky, good to know! 👌